library/src/hcc_detail/hipsolver.cpp

```diff
@@ -697,6 +714,219 @@ catch(...)
     return exception2hip_status();
 }
 
+/******************** GESVD ********************/
+hipsolverStatus_t hipsolverSgesvd_bufferSize(hipsolverHandle_t handle, int m, int n, int* lwork)
+try
+{
+    *lwork = 0;
+    return HIPSOLVER_STATUS_NOT_SUPPORTED;
+}
+catch(...)
+{
+    return exception2hip_status();
+}
+
+hipsolverStatus_t hipsolverDgesvd_bufferSize(hipsolverHandle_t handle, int m, int n, int* lwork)
+try
+{
+    *lwork = 0;
+    return HIPSOLVER_STATUS_NOT_SUPPORTED;
+}
+catch(...)
+{
+    return exception2hip_status();
+}
+
+hipsolverStatus_t hipsolverCgesvd_bufferSize(hipsolverHandle_t handle, int m, int n, int* lwork)
+try
+{
+    *lwork = 0;
+    return HIPSOLVER_STATUS_NOT_SUPPORTED;
+}
+catch(...)
+{
+    return exception2hip_status();
+}
+
+hipsolverStatus_t hipsolverZgesvd_bufferSize(hipsolverHandle_t handle, int m, int n, int* lwork)
+try
+{
+    *lwork = 0;
+    return HIPSOLVER_STATUS_NOT_SUPPORTED;
+}
+catch(...)
+{
+    return exception2hip_status();
+}
+
+hipsolverStatus_t hipsolverSgesvd(hipsolverHandle_t handle,
+                                  signed char       jobu,
+                                  signed char       jobv,
+                                  int               m,
+                                  int               n,
+                                  float*            A,
+                                  int               lda,
+                                  float*            S,
+                                  float*            U,
+                                  int               ldu,
+                                  float*            V,
+                                  int               ldv,
+                                  float*            work,
+                                  int               lwork,
+                                  float*            rwork,
+                                  int*              devInfo)
+try
+{
+    if(!rocblas_is_managing_device_memory((rocblas_handle)handle))
+        rocblas_set_workspace((rocblas_handle)handle, nullptr, 0);
+
+    return rocblas2hip_status(rocsolver_sgesvd((rocblas_handle)handle,
+                                               char2rocblas_svect(jobu),
+                                               char2rocblas_svect(jobv),
+                                               m,
+                                               n,
+                                               A,
+                                               lda,
+                                               S,
+                                               U,
+                                               ldu,
+                                               V,
+                                               ldv,
```

Comment:
You should set the workspace to user-owned (rocblas_set_workspace())...
Default to auto managing only if work==nullptr as with the other functions accepting workspace provided by user.

Reply:
I'm planning to do that if/when we support the bufferSize function for gesvd. Otherwise, there's a risk that the user could provide a workspace that is not sufficiently large for gesvd and cause a memory fault.